packages/iov-bcp-types/types/connection.d.ts

```diff
@@ -154,6 +185,12 @@ export interface BcpConnection {
     /** @deprecated use watchBlockHeaders().map(header => header.height) */
     readonly changeBlock: () => Stream<number>;
     readonly postTx: (tx: PostableBytes) => Promise<PostTxResponse>;
+    /**
+     * Looks up transaction in history and if not found, waits until it is available.
+     *
+     * As a consequence, this never resolves for non-existing transaction IDs.
+     */
+    readonly waitForTransaction?: (id: TransactionId) => Promise<ConfirmedTransaction | FailedTransaction>;
```

Comment:
This interface has one significant flaw: there is not way to cancel the waiting externally. If we use it as part of a stream producer (on postTx for example), we cannot stop this, even if noone listens to the stream anymore.

So we might be forced to convert it into a single event stream, which allows cancellation.